cmd/jet/main.go

```diff
@@ -66,17 +71,28 @@ func init() {
 	flag.StringVar(&schemaName, "schema", "public", `Database schema name. (default "public")(PostgreSQL only)`)
 	flag.StringVar(&params, "params", "", "Additional connection string parameters(optional). Used only if dsn is not set.")
 	flag.StringVar(&sslmode, "sslmode", "disable", `Whether or not to use SSL. Used only if dsn is not set. (optional)(default "disable")(PostgreSQL only)`)
-	flag.StringVar(&ignoreTables, "ignore-tables", "", `Comma-separated list of tables to ignore`)
-	flag.StringVar(&ignoreViews, "ignore-views", "", `Comma-separated list of views to ignore`)
-	flag.StringVar(&ignoreEnums, "ignore-enums", "", `Comma-separated list of enums to ignore`)
+	flag.StringVar(&ignoreTables, "ignore-tables", "", `Comma-separated list of tables to ignore.`)
+	flag.StringVar(&ignoreViews, "ignore-views", "", `Comma-separated list of views to ignore.`)
+	flag.StringVar(&ignoreEnums, "ignore-enums", "", `Comma-separated list of enums to ignore.`)
 
-	flag.StringVar(&destDir, "path", "", "Destination dir for files generated.")
-}
+	flag.StringVar(&destDir, "path", "", "Destination directory for files generated.")
+	flag.StringVar(&modelPkg, "model-pkg", "model", "Relative path for the Model files package from the destination directory.")
+	flag.StringVar(&tablePkg, "table-pkg", "table", "Relative path for the Table files package from the destination directory.")
+	flag.StringVar(&viewPkg, "view-pkg", "view", "Relative path for the View files package from the destination directory.")
+	flag.StringVar(&enumPkg, "enum-pkg", "enum", "Relative path for the Enum files package from the destination directory.")
 
-func main() {
 	flag.Usage = usage
 	flag.Parse()
 
+	// Convert the OS-specific path separator to slashes for cross-platform compatibility.
+	destDir = filepath.ToSlash(destDir)
+	modelPkg = filepath.ToSlash(modelPkg)
+	tablePkg = filepath.ToSlash(tablePkg)
+	viewPkg = filepath.ToSlash(viewPkg)
+	enumPkg = filepath.ToSlash(enumPkg)
```

Comment:
This wouldn't be needed if the following method in the `Model` struct used `filepath.Base()` instead of `path.Base()`,

```go
// model_template.go:24
// PackageName returns package name of model types
func (m Model) PackageName() string {
	return path.Base(m.Path)
}
```

The `path` package doesn't work correctly with backslashes ([source](https://pkg.go.dev/path))

> Package path implements utility routines for manipulating slash-separated paths.
>
>The path package should only be used for paths separated by forward slashes, such as the paths in URLs. This package does not deal with Windows paths with drive letters or backslashes; to manipulate operating system paths, use the [path/filepath](https://pkg.go.dev/path/filepath) package.

And almost all the path-related operations (those I've seen at least) in Jet are done using the `path` package.

\
To illustrate the problem,
```go
fmt.Println(path.Base( ".\\entity"))             // .\entity
fmt.Println(path.Base(".\\test\\entity"))       // .\test\entity
fmt.Println(path.Base("./entity"))             // entity
```

Whereas all of these paths correctly return `entity` as the base when using the `filepath.Base()` function.
\
As a result of this, when Jet is invoked with the model path flag set to `.\entity` on Windows, the generated model's package statement becomes `package .\entity` because it's getting the package name from the `PackageName()` method in the `process.go:processTableModels()` function,
```go
text, err := generateTemplate(
	// ...
	template.FuncMap{
		"package": func() string {
			return modelTemplate.PackageName()
		},
		// ...
})
```

And the `format.Source(text)` call in `filesys.go:FormatAndSaveGoFile()` fails because of the invalid package name.

\
Admittedly my change is more of a workaround. A better approach would be to migrate all the file path related operations from using the `path` package to the `filepath` package.

Reply:
I think changing all the usage  of the `path` package to the `filepath` package is the right approach.
I've looked into all the usage of `path` and it should be relatively simple and non breaking.

Shall I go along with that?

Reply:
As it turns out, there were actually not that many `path` package usage. So I fixed them.
Works on Windows without the replacing slash workaround now.

Reply:
> I think changing all the usage of the path package to the filepath package is the right approach.
I've looked into all the usage of path and it should be relatively simple and non breaking.

Yeah, it makes sense.